src/shork_ffi.erl

```diff
@@ -62,9 +62,15 @@ transaction(#shork_connection{pid = Pid} = Conn, Callback) ->
       {error, {transaction_rolled_back, Reason}}
   end.
 
-query(#shork_connection{pid = Pid}, Sql, Arguments) ->
-  Res = mysql:query(Pid, Sql, Arguments),
-  case Res of
+query(#shork_connection{pid = Pid, default_timeout = DefaultTimeout}, Sql, Arguments, Timeout) ->
+  Timeout1 =
+    case Timeout of
+      none ->
+        DefaultTimeout;
+      {some, QueryTimeout} ->
+        QueryTimeout
+    end,
```

Comment:
Maybe a name like `ActualTimeout` would be better. Or even better: The parameter is renamed to OptionalTimeout or something like that.